tests/reference/c-print_01-4d44628.stdout

```diff
@@ -40,9 +40,12 @@ void f()
     char * x = (char*) malloc(40 * sizeof(char));
     char * y = (char*) malloc(40 * sizeof(char));
     printf("%s\n", "Hello World!");
+    if (!x) x = (char *) malloc(40 * sizeof(char));
     strcpy(x, ",");
+    if (!y) y = (char *) malloc(40 * sizeof(char));
```

Comment:
Actually here (i.e., `strcpy(x, ",")`), `x` needs to be pre-allocated. We do this right now by doing `malloc(40, sizeof(char))`. This means we are allocating a string size of 40 and assigning it to `x`. This is simple but sometimes means wastage of memory and sometimes means chances of segmentation fault. So a better way is to define our own `_lfortran_strcpy` in `lfortran_intrinsics.h` and then call C's `strcpy` in it. Then, our `_lfortran_strcpy` will be called everywhere. This is how it will look like,

```c
void _lfortran_strcpy(char* x, char* y) {
    x = (char*) malloc((strlen(x) + strlen(y) + 1) * sizeof(char));
    strcpy(x, y);
}
```

Reply:
@Smit-create Please do the above in this PR only. I think the above approach is better than what we are doing right now.